crates/matrix-sdk/src/send_queue/mod.rs

```diff
@@ -618,7 +638,23 @@ impl RoomSendQueue {
                 continue;
             };
 
-            match Self::handle_request(&room, queued_request, cancel_upload_rx).await {
+            // Prepare to watch and communicate the request progress for media uploads.
+            let media_upload_info =
+                RoomSendQueue::try_create_media_upload_info(&queued_request, &room, &queue)
+                    .await
+                    .unwrap_or_default();
+            let progress = RoomSendQueue::try_create_media_upload_progress_observable(
+                &report_media_upload_progress,
+                &media_upload_info,
+                &related_txn_id,
+                room_id,
+                &global_update_sender,
+                &update_sender,
+            );
```

Comment:
This code will always create a progress listener, even for sending non-medias. I don't think we should do this, so maybe it is time to split the code in such a way that sending a plain event or a media go through different code paths. Let's not act on it quite yet; i'd like to think more about this.